### DISCUSSION TITLE
Editing source code

### DISCUSSION THREAD

**Discussion by NiSlad** [Universite-Gustave-Eiffel/NoiseModelling#655] [Q&A] [answered]
Hi

I have modified some coefficients in the "RoadCnossos" code and tried to create an executable using [this](https://github.com/Nitnelav/NoiseModelling-installer) tutorial.  Unfortunately despite successfully creating the installer and running it the installed application does not run. 

I found other users with similar question but none of the tutorial links work anymore so I was wondering if someone has a solution or any advice?

**Comment by nicolas-f** [comment]
Hi
You can install an old version and replace the NoiseModelling subfolder. The exe is just a call to the java.exe program.

**Reply by NiSlad** [reply]
Could you please explain a little more? The NoiseModelling_4.0.5 subfolder in the directory where I installed the programme does not have .json file with the code that I needed to change so I don't know what I should replace it with.

**Reply by nicolas-f** [reply, marked as answer]
Oh sorry you talked about creating the exe but the previous step was to compile NoiseModelling.

You should fork NoiseModelling on github then modify the json file on the[ 4.X branch](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/blob/4.X/)

When you have commited your modifications github action will build NoiseModelling. You can then download this file:

NoiseModelling_without_gui.zip 
https://github.com/NiSlad/NoiseModelling/actions/workflows/CI.yml

The emission json file are in the file named **noisemodelling-emission-4.0.5.jar**

So you have to copy this file:

lib\noisemodelling-emission-4.0.5.jar
to this directory
NoiseModelling_4.0.5\webapps\geoserver\WEB-INF\lib\noisemodelling-emission-4.0.5.jar

**Reply by NiSlad** [reply]
Works exactly as I needed it too. Thank you so much!

**Comment by atenie** [comment]
Hi, late bump to the thread, but I assume it's a good place to expand on how the whole thing works.

Now the software is on version 5. I assume we have to fork the main branch.
Say I modified noisemodelling-jdbc-5.0.**1**, which generates me a noisemodelling-jdbc-5.0.**2**.jar. Say I want to test it in the GUI version, for ease of use.

Using the normal 5.0.2 jar file in that path (NoiseModelling_x.y.z\webapps\geoserver\WEB-INF\lib\) doesn't work for GUI. I saw that there's a start.jar file in the home path of the folder, how do I .jar it for the GUI version?
